lib/instance.js

```diff
@@ -236,15 +236,14 @@ module.exports = (function() {
 
         if (options.attributes) {
           keys = options.attributes;
-          if (this.Model._hasVirtualAttributes) {
-            keys = keys.concat(this.Model._virtualAttributes);
-          }
 
           if (this.options.includeNames) {
             keys = keys.concat(this.options.includeNames);
+          } else if (this.Model._hasVirtualAttributes) {
+            keys = keys.concat(this.Model._virtualAttributes);
           }
 
-          for (i = 0, length = keys.length; i < length; i++) {
+          for (i = 0; i < keys.length; i++) {
```

Comment:
This is not true anymore. V8 will optimize this. As backend developers we do not have to support old browsers :)